### ISSUE TITLE
Regex filter does not allow to filter for ProductIdentifier "1234567-abcdefgh"

### ISSUE THREAD

**Issue by dacky179** [PHOENIXCONTACT/MORYX-AbstractionLayer#225]
**Describe the bug**
Our application does not have unique productidentifier. We have a combination of productidentifier and configurationkey.
After creation of the product with such a filter I mention 2 faulti behaviors:
- If I want to find the product in the ui by typing 1234567-abcdefgh i get no results.
- If I want to import new products i surely check if same product already exists. if a product with identifier "1234567" exists it will match to the check for "1234567-abcdefgh".

This behavior is caused by a regex check in ProductStorage.LoadTypes().
https://github.com/PHOENIXCONTACT/MORYX-AbstractionLayer/blob/db3eeb42343e8c8c6eea5c55232bb1a334df784a/src/Moryx.Products.Management/Implementation/Storage/ProductStorage.cs#L330C8-L330C9

**Expected behavior**
I expected to find my product by exact identifier match

**Screenshots**
![image](https://github.com/PHOENIXCONTACT/MORYX-AbstractionLayer/assets/67679872/1141d1e8-c24a-4067-afc2-db3f7b1b1652)
Existing Article:
![image](https://github.com/PHOENIXCONTACT/MORYX-AbstractionLayer/assets/67679872/2db84797-12a2-42e1-ab2b-5af5a5adc5e3)
Filter for that article:
![image](https://github.com/PHOENIXCONTACT/MORYX-AbstractionLayer/assets/67679872/7dc70658-83a4-40a1-8beb-e2361eb57e3a)

**Environment:**
 - OS: Windows10
 - Version 5.15.6

**Additional context**
I think It was added to allow contains, startwith, endwith and equals search with wildcards from UI.
I would prefer to allow change filter expression to be all but \*.
I future I would suggest to move that funtiction to UI side:
let the user decide to search with regex or simplified with * wildchard.
if he choose simplified, the ui code replaces the * with a regex wildcard '.*'
the api always handles the entered string as a regex.

**Comment by 1nf0rmagician** [comment]
@Toxantron, @dbeuchler is it intentional, that the regex for the filter part in the pattern does not include the `-` character. I think nowhere else is this constraint enforced is it?

**Comment by dbeuchler** [comment]
> @Toxantron, @dbeuchler is it intentional, that the regex for the filter part in the pattern does not include the `-` character. I think nowhere else is this constraint enforced is it?

No, I think we just didn't think about this case. I like the idea of @dacky179 to support regex at all
